### ISSUE TITLE
Config written incorrectly, when calling _remove twice()

### ISSUE THREAD

**Issue by mmontagna** [tmont/nginx-conf#6]
For example, given something like the following config file:

```nginx
http {
    test one {
        exist;
    }
    test2 two {
        exist;
    }
}
```

If I try and call *._remove on "test" and "test2" I can get strange results written to my nginx.conf file. Sometimes only one property is removed, sometimes the results get jumbled. It looks like writes are being intermingled. 

Is this the expected behavior? I don't see this documented anywhere. It's possible to avoid this issue by calling flush() with a callback and chaining all nginx-conf calls that way, but this is cumbersome and the README seems to imply that this shouldn't be necessary.

**Comment by mmontagna** [comment]
I submitted a pull request.  I'm not 100% sure it's the right thing to do, but it which fixes the issue for me.

**Comment by tmont** [comment]
Hmm. It sounds like the writes are definitely mingling in a way that's not awesome. I think the preferred solution would just be to buffer/batch the writes.

Right now it's event-driven, so probably the best course of action would be to make the `flush()` method a little smarter, about handling consecutive writes.

The PR fixes the issue, but is still issuing multiple writes, which I don't really like. Something like this would be best:

```javascript
var timeout;
function write(data) {
  if  (timeout) {
    clearTimeout(timeout);
  }

  timeout = setTimeout(function() {
    //last write wins
    fs.writeFile('/the/file', data, ...);
  }, 100);
}
```

I don't remember off the top of my head if employing a "last write wins" strategy would cause issues.

**Comment by mmontagna** [comment]
OK, thanks! I'm not very familiar with this project, but I'd guess that last write wins makes sense. I'll try it out this evening.

**Comment by mmontagna** [comment]
OK. I just send another PR which should batch writes.

[https://github.com/tmont/nginx-conf/pull/8]

**Comment by tmont** [comment]
fixed in npm: `nginx-conf@1.1.0`
